tests/Generator/Resolver/Value/Chainable/FixtureMethodCallReferenceResolverTest.php

```diff
@@ -0,0 +1,215 @@
+    {
+        $resolver = new FixtureMethodCallReferenceResolver();
+        $newResolver = $resolver->withValueResolver(new FakeValueResolver());
+
+        $this->assertEquals(new FixtureMethodCallReferenceResolver(), $resolver);
+        $this->assertEquals(new FixtureMethodCallReferenceResolver(new FakeValueResolver()), $newResolver);
+    }
+
+    public function testCanResolveMethodCallReferenceValues()
+    {
+        $resolver = new FixtureMethodCallReferenceResolver();
+
+        $this->assertTrue($resolver->canResolve(new FixtureMethodCallValue(new FakeValue(), new FunctionCallValue('method'))));
+        $this->assertFalse($resolver->canResolve(new FakeValue()));
+    }
+
+    /**
+     * @expectedException \Nelmio\Alice\Throwable\Exception\Generator\Resolver\ResolverNotFoundException
+     * @expectedExceptionMessage Expected method "Nelmio\Alice\Generator\Resolver\Value\Chainable\FixtureMethodCallReferenceResolver::resolve" to be called only if it has a resolver.
+     */
+    public function testCannotResolveValueIfHasNoResolver()
+    {
+        $value = new FixtureMethodCallValue(new FakeValue(), new FunctionCallValue('method'));
+        $resolver = new FixtureMethodCallReferenceResolver();
+        $resolver->resolve($value, new FakeFixture(), ResolvedFixtureSetFactory::create(), [], new GenerationContext());
+    }
+
+    public function testReturnsSetWithResolvedValue()
+    {
+        $value = new FixtureMethodCallValue(
+            $reference = new FakeValue(),
+            $functionCall = new FunctionCallValue('getFoo', [
+                $arg1 = new FakeValue(),
+            ])
+        );
+        $fixture = new FakeFixture();
+        $set = ResolvedFixtureSetFactory::create(new ParameterBag(['foo' => 'bar']));
+        $scope = ['val' => 'scopie'];
+        $context = new GenerationContext();
+        $context->markIsResolvingFixture('foo');
+
+        $valueResolverContext = new GenerationContext();
+        $valueResolverContext->markIsResolvingFixture('foo');
+        $valueResolverContext->markAsNeedsCompleteGeneration();
+
+        $dummyProphecy = $this->prophesize(DummyWithGetter::class);
+        $dummyProphecy->getFoo('resolved_argument')
+            ->shouldBeCalled()
+            ->willReturn('resolved_value');
+
+        $valueResolverProphecy = $this->prophesize(ValueResolverInterface::class);
+        $valueResolverProphecy
+            ->resolve($arg1, $fixture, $set, $scope, $context)
+            ->willReturn(
+                new ResolvedValueWithFixtureSet(
+                    'resolved_argument',
+                    $newSet = ResolvedFixtureSetFactory::create(new ParameterBag(['ping' => 'pong']))
+                )
+            )
+        ;
+        $valueResolverProphecy
+            ->resolve($reference, $fixture, $newSet, $scope, $valueResolverContext)
+            ->willReturn(
+                new ResolvedValueWithFixtureSet(
+                    $instance = $dummyProphecy->reveal(),
```

Comment:
I prefer the:

```php
/** @var DummyWithGetter $dummy */
$dummy = $dummyProphecy->reveal();
```

convention as you don't have typehint warnings for the services with static analysis tools